go/vt/discovery/healthcheck_test.go

```diff
@@ -505,6 +505,16 @@ func TestHealthCheckCloseWaitsForGoRoutines(t *testing.T) {
 	assert.Nil(t, hc.healthByAlias, "health data should be nil")
 }
 
+// TestHealthCheckStartCloseRace tests that rapidly calling Start()/Close()
+// still cleanly shuts down.
+func TestHealthCheckStartCloseRace(t *testing.T) {
+	ts := memorytopo.NewServer("cell")
+	hc := createTestHc(ts)
+	assert.Nil(t, hc.Close(), "Close returned error")
+	ts.Close()
+	time.Sleep(50 * time.Millisecond)
```

Comment:
I don't get it. why sleep at the end of the method like this?

Reply:
Without it, the race/NPE was not triggered. Basically, we need to give the health-check time to actually run the goroutine on a closed topo-server in order to get an NPE. This test-case is for a super-niche bug, so I'd be fine removing if you want.

Reply:
Is there a way we can add an assertion to assert the health check is correctly shut down on line 515?

Reply:
note that the `time.Sleep` might be a bit sensitive as it can introduce flakiness in the CI builds